src/compiler/emitter.ts

```diff
@@ -2468,18 +2608,35 @@ var __param = this.__param || function(index, decorator) { return function (targ
                 if (node.flags & NodeFlags.Export) {
                     writeLine();
                     emitStart(node);
-                    if (node.flags & NodeFlags.Default) {
-                        if (languageVersion === ScriptTarget.ES3) {
-                            write("exports[\"default\"]");
-                        } else {
-                            write("exports.default");
+
+                    if (compilerOptions.module === ModuleKind.System) {
+                        // emit export default <smth> as
+                        // export("default", <smth>)
+                        write(`${exportFunctionForFile}("`);
+                        if (node.flags & NodeFlags.Default) {
+                            write("default");

```

Comment:
should also handle the ES3 case and write ["default"]. also can you add a test for it.